content/guide/plugins-authoring.md

```diff
@@ -298,7 +298,65 @@ def start(**kwargs):
 
 # Testing Your Plugin
 
-In most cases, the recommendation is to test your plugin's logic using local workflows and only then, run them as part of a Cloudify [deployment]({{page.terminology_link}}#deployment). The [Plugin Template]({{page.template_link}}/blob/3.2/plugin/tests/test_plugin.py) has an example of doing just that.
+In most cases, the recommendation is to test your plugin's logic using local workflows and only then, run them as part of a Cloudify [deployment]({{page.terminology_link}}#deployment). We have supplied you with a nice and tidy
+decorator to do just that. It is provided by the `cloudify-plugins-common.test_utils`, and it's very intuitive to use, But just in case let's look at an example:
+
+{{< gsHighlight  python >}}
+from cloudify.test_utils import workflow_test
+
+@workflow_test(
+                blueprint_path,
+                copy_plugin_yaml,
+                resources_to_copy,
+                temp_dir_prefix,
+                init_args,
+                inputs
+                )
+def test_my_task(self, cfy_local):
+    pass
+{{< /gsHighlight >}}
+
+### Now lets break down the arguments:
+- blueprint_path - A path to the blueprint to run - This is the ONLY mandatory input.
+- copy_plugin_yaml - Tries to locate and copy the plugin.yaml file to the temporary test dir created.
+- resources_to_copy - If any additional resources are needed for the test, they could be supplied here via list.
+- temp_dir_prefix - If you have any special request for the temprary test dir prefix, you should supply it here.
```

Comment:
"for THE temporary test dir" - is it that clear that there is one in the first place? what is it used for? why would i want to control its prefix? what benefit do i gain from it?